test/test_common/utility_core.cc

```diff
@@ -0,0 +1,35 @@
+#include "common/common/thread_impl.h"
```

Comment:
Are these two extracted for avoid circular dependency? The "utility_core" vs "utility" split sounds confusing. Any idea for a better naming? "filesystem.h" and "thread.h"?

Reply:
Exactly - split for circular dependency. Yes I can split them.

Reply:
OK let's do that, other than this the PR LGTM.

Reply:
done.